specs/casper_sharding_v2.1.md

```diff
@@ -581,27 +581,43 @@ Finally:
 * Let `next_start_shard = (shard_and_committee_for_slots[-1][-1].shard_id + 1) % SHARD_COUNT`
 * Set `shard_and_committee_for_slots[CYCLE_LENGTH:] = get_new_shuffling(block.ancestor_hashes[0], validators, next_start_shard)`
 
--------
-
-Note: this is ~80% complete. The main sections that are missing are:
-
-* Logic for the formats of shard chains, who proposes shard blocks, etc. (in an initial release, if desired we could make crosslinks just be Merkle roots of blobs of data; in any case, one can philosophically view the whole point of the shard chains as being a coordination device for choosing what blobs of data to propose as crosslinks)
-* Logic for inducting queued validators from the PoW chain
-* Penalties for signing or attesting to non-canonical-chain blocks (update: may not be necessary, see https://ethresear.ch/t/attestation-committee-based-full-pos-chains/2259)
-* Per-validator proofs of custody, and associated slashing conditions
-* Versioning and upgrades
-
-Slashing conditions may include:
-
-
-    Casper FFG slot equivocation [done]
-    Casper FFG surround [done]
-    Beacon chain proposal equivocation [done]
-    Shard chain proposal equivocation
-    Proof of custody secret leak
-    Proof of custody wrong custody bit
-    Proof of custody no secret reveal
-    RANDAO leak
+### TODO
+
+Note: This spec is ~60% complete.
+
+* [ ] Specify how `crystallized_state_root` and `active_state_root` are constructed
+* [ ] Specify the shard chain blocks, blobs, proposers, etc.
+* [ ] Specify the rules for forced deregistrations
+* [ ] Add a double-batched Merkle accumulator for historical beacon chain blocks
+* [ ] Specify the various assumptions (global clock, validator honesty, validator liveness, etc.)
+* [ ] Specify in Solidity the registration contract on the PoW chain
```

Comment:
Because Solidity is more accessible? I think we can keep using Vyper and just need to share the bytecode and ABIs for tests.

Reply:
Sorry, I meant Vyper. Will edit 👍